tests/integration/ceph_base_file_test.go

```diff
@@ -287,6 +287,28 @@ func runFileE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.S
 	assertNoErrorUnlessNotFound(s, err)
 }
 
+func runPreserveFsOnDeleteTest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.Suite, namespace string) {
+	logger.Infof("Preserve Fs On Delete Smoke Test")
+
+	fsName := "testfs"
+	activeCount := 2
+	createFilesystem(helper, k8sh, s, namespace, fsName, activeCount)
+
+	_, err := k8sh.Kubectl("-n", namespace, "patch", "CephFilesystem", fsName, "--type=merge", "-p", `{"spec": {"preserveFsOnDelete": true}}`)
+	assert.NoError(s.T(), err)
+
+	clusterInfo := client.AdminClusterInfo(namespace)
+	err = waitForFilesystemActive(k8sh, clusterInfo, fsName)
+	assert.NoError(s.T(), err)
+
+	cleanupFilesystem(helper, k8sh, s, namespace, fsName)
+
+	fses, err := helper.FSClient.List(namespace)
+	assert.NoError(s.T(), err)
+
+	require.Equal(s.T(), []string{fsName}, fses)
```

Comment:
The test will need to clean up the filesystem in case another test tries to create a filesystem after this. The same rook cluster is used for multiple tests.